### PR TITLE
Use updated ctap-types with references

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
This PR uses the updated ctap-types crate that uses references rather than heapless buffers to reduce the total stack usage. See https://github.com/Nitrokey/ctap-types/pull/11

### CODE REVIEW

File: src/lib/types.rs
Comment (robin-nitrokey) [inline src/lib/types.rs]: Nit: I think `[1..(4 + 1)]` is a bit hard to read, I’d either use `[1..5]` or, if you want to highlight that the lengths match, `[1..][..4]`.

Not related to this PR, but it looks like there is a check for `arr.size()` missing here?

**Review by robin-nitrokey — APPROVED** [review]

File: src/lib/types.rs
Comment (sosthene-nitrokey) [reply src/lib/types.rs]: I was surprised that I didn't use `[1..][..4]`, and looking at the history it looks like it came from a `cargo clippy --fix`.

Will make a clippy issue to see if that  can be improved.

**Comment by sosthene-nitrokey** [comment]
I can't merge this or add reviewer. @szszszsz can you add the core team to the maintainers?

File: src/lib/types.rs
Comment (sosthene-nitrokey) [reply src/lib/types.rs]: It also added `wc_magic_number[..4]` that was unnecessary.

**Pull request merged**
